tt_metal/impl/buffers/buffer.hpp

```diff
@@ -265,6 +268,7 @@ class Buffer {
     BufferType buffer_type_;
     TensorMemoryLayout buffer_layout_;
     std::optional<ShardSpecBuffer> shard_parameters_;
+    bool is_allocated_;
```

Comment:
```suggestion
    bool is_allocated_ = false;
```

Buffer has multiple constructors. 
It looks like this variable is only initialized in one of them.
Going forward please initialize primitive variables on declaration to avoid undefined behavior